src/registrar/forms/domain_request_wizard.py

```diff
@@ -22,10 +22,151 @@
+            )
+
+    def clean_sub_organization(self):
+        """On suborganization clean, set the suborganization value to None if the user is requesting
+        a custom suborganization (as it doesn't exist yet)"""
+
+        # If it's a new suborganization, return None (equivalent to selecting nothing)
+        if self.cleaned_data.get("is_requesting_new_suborganization"):
+            return None
+
+        # Otherwise just return the suborg as normal
+        return self.cleaned_data.get("sub_organization")
+
+    def full_clean(self):
+        """Validation logic to remove the custom suborganization value before clean is triggered.
+        Without this override, the form will throw an 'invalid option' error."""
+        # Remove the custom other field before cleaning
+        data = self.data.copy() if self.data else None
+
+        # Remove the 'other' value from suborganization if it exists.
+        # This is a special value that tracks if the user is requesting a new suborg.
+        suborganization = self.data.get("portfolio_requesting_entity-sub_organization")
+        if suborganization and "other" in suborganization:
+            data["portfolio_requesting_entity-sub_organization"] = ""
+
+        # Set the modified data back to the form
+        self.data = data
+
+        # Call the parent's full_clean method
+        super().full_clean()
+
+    def clean(self):
+        """Custom clean implementation to handle our desired logic flow for suborganization.
+        Given that these fields often rely on eachother, we need to do this in the parent function."""
+        cleaned_data = super().clean()
+
+        # Do some custom error validation if the requesting entity is a suborg.
+        # Otherwise, just validate as normal.
+        suborganization = self.cleaned_data.get("sub_organization")
+        is_requesting_new_suborganization = self.cleaned_data.get("is_requesting_new_suborganization")
+
+        # Get the value of the yes/no checkbox from RequestingEntityYesNoForm.
+        # Since self.data stores this as a string, we need to convert "True" => True.
+        requesting_entity_is_suborganization = self.data.get(
+            "portfolio_requesting_entity-requesting_entity_is_suborganization"
+        )
+        if requesting_entity_is_suborganization == "True":
+            if is_requesting_new_suborganization:
+                # Validate custom suborganization fields
+                if not cleaned_data.get("requested_suborganization"):
+                    self.add_error("requested_suborganization", "Enter details for your organization name.")
+                if not cleaned_data.get("suborganization_city"):
+                    self.add_error("suborganization_city", "Enter details for your city.")
+                if not cleaned_data.get("suborganization_state_territory"):
+                    self.add_error("suborganization_state_territory", "Enter details for your state or territory.")
```

Comment:
**blocking** Why don't these errors match the ones you have above in the field definitions? If the others aren't triggering due to the setup, they should be removed to avoid confusion and the content copied here.

Reply:
Q: Why aren't we using error definitions on the form fields instead?